File: python/reverse_int.py
#!/usr/bin/env python
def reverse_int(number):
    x, res = abs(number), 0
    sign = ((1 << 31) - 1) if number > 0 else (1 << 31)
    while x != 0:
        res = res * 10 + x % 10
        if res > sign:
            return 0
        x //= 10
    return res if number > 0 else -res


print(reverse_int(int(str(1 << 31)[::-1])))
print(reverse_int(int(str((1 << 31) - 1)[::-1])))
print(reverse_int(-int(str(1 << 31)[::-1])))
print(reverse_int(-int(str((1 << 31) - 1)[::-1])))
print(reverse_int(-int(str((1 << 31) + 1)[::-1])))

File: typescript/reverse_int.ts
function reverseInt(num: number): number {
    let compareValue: number = num < 0 ? 2**31 : 2**31 - 1;
    let x: number = Math.abs(num);
    let res: number = 0;
    while(x != 0) {
        res = res * 10 + Math.floor(x % 10);
        if(res > compareValue) {
            return 0;
        }
        x = Math.floor(x/10);

    }
    return num < 0 ? -res : res
}
console.log(reverseInt(+(2**31+1).toString().split('').reverse().join('')))
console.log(reverseInt(+(2**31).toString().split('').reverse().join('')))
console.log(reverseInt(+(2**31-1).toString().split('').reverse().join('')))
console.log(reverseInt(-+(2**31).toString().split('').reverse().join('')))
console.log(reverseInt(-+(2**31+1).toString().split('').reverse().join('')))